server/src/main/java/org/opensearch/plugins/PluginInfo.java

```diff
@@ -417,8 +431,17 @@ public String getFolderName() {
      *
      * @return the names of the plugins extended
      */
+    public boolean isExtendedPluginOptional(String extendedPlugin) {
+        return optionalExtendedPlugins.contains(extendedPlugin);
+    }
+
+    /**
+     * Other plugins this plugin extends through SPI
+     *
+     * @return the names of the plugins extended
+     */
     public List<String> getExtendedPlugins() {
-        return extendedPlugins;
+        return extendedPlugins.stream().map(s -> s.split(";")[0]).collect(Collectors.toUnmodifiableList());
```

Comment:
Why do we need that? `extendedPlugins` would never contain `;`, only plugin name

Reply:
Removed this here. `extendedPlugins` will now have all regardless of optionality, but there would be another list that contains only the optional. Tested the Nodes Info API and confirmed its being returned:

```
...
"plugins" : [
  {
    "name" : "opensearch-index-management",
    "version" : "3.0.0.0-SNAPSHOT",
    "opensearch_version" : "3.0.0",
    "java_version" : "21",
    "description" : "OpenSearch Index Management Plugin",
    "classname" : "org.opensearch.indexmanagement.IndexManagementPlugin",
    "custom_foldername" : "",
    "extended_plugins" : [
      "opensearch-job-scheduler"
    ],
    "has_native_controller" : false,
    "optional_extended_plugins" : [
      "opensearch-job-scheduler"
    ]
  }
]
...
```